lib/internal/fs/streams.js

```diff
@@ -113,7 +137,8 @@ Object.setPrototypeOf(ReadStream.prototype, Readable.prototype);
 Object.setPrototypeOf(ReadStream, Readable);
 
 ReadStream.prototype.open = function() {
-  fs.open(this.path, this.flags, this.mode, (er, fd) => {
+  const open = this[kOpen] || fs.open;
```

Comment:
Can we undo this now and always set `this[kOpen]` etc. in the constructor?

(Tbh, I still don’t see why we wouldn’t just set `this[kFs]` to either `options.fs || fs` and use that, but that’s not a hill I’m going to die on…)

Reply:
It seems like the only reason for the constructor to store references to the individual methods is to ensure the stream does not honor monkey-patching performed after construction, but unless the constructor takes copies of the native `fs` methods this seems like it would be ineffective?

Reply:
@addaleax: Because we need to honor monkey patching `fs´ after construction. Otherwise I think this needs to be a semver-major.

I didn't follow @coreyfarrell?

Reply:
@ronag Take the following two calls to `fs.createReadStream`:
```js
const fs = require('fs');
const stream1 = fs.createReadStream(__filename, {fs});
const stream2 = fs.createReadStream(__filename);
fs.read = () => {
  throw new Error('hit monkey-patch code');
};
```
I thought the idea was to ignore monkey-patching after construction regardless of having an `fs` option or not.
I had assumed that `stream1` and `stream2` should behave the same, either both hitting the throw or not.  I understand the semver-major argument though so I defer to others on this.

Reply:
Yes, I agree, we should honour monkey-patching after construction, and that that is the less breaking variant here. Others seem to disagree, though.

Reply:
I'm fine with either. For me it's more of a question of semver minor vs major.